tests/TestCase/ORM/CompositeKeysTest.php

```diff
@@ -254,6 +256,10 @@ public function internalStategiesProvider()
      */
     public function testBelongsToEager($strategy)
     {
+        $assoc = new BelongsTo('Test');
+        if (!$assoc->validStrategy($strategy)) {
```

Comment:
I would have different providers for the different association types to avoid conditionals in the test code.